### ISSUE TITLE
OrderBy / ThenBy sorts strings by their character index rather than case insensitively by default

### ISSUE THREAD

**Issue by Porl91** [bridgedotnet/Bridge#3477]
When sorting a collection of strings the output in Bridge differs from .NET, as in Bridge the strings are ordered in relation to the index of the characters, rather than the strings being compared case insensitively, as in .NET. This affects the **OrderBy** / **OrderByDescending** / **ThenBy** / **ThenByDescending** methods.

For example, 
In C#:
`"A", "a", "b", "B"` becomes:
`"a", "A", "b", "B"`

In Bridge:
`"A", "a", "b", "B"`  becomes:
`"A", "B", "a", "b""`

### Steps To Reproduce

C#:
https://dotnetfiddle.net/6zvutJ

Bridge:
https://deck.net/9faed830fa73f8b3d48c2b8e921d67cd

### Expected Result

```
a - A - b - B
```

### Actual Result

```
A - B - a - b
```

As a workaround I've just created a custom **IComparer** that I pass into these sorting methods when I'm sorting case insensitively, as I see that there isn't a **CaseInsensitiveComparer** class.

https://msdn.microsoft.com/en-us/library/system.collections.caseinsensitivecomparer(v=vs.110).aspx

**Comment by fabriciomurta** [comment]
Hello @Porl91!

Thank you very much for reporting the issue, we've just implemented logic in Bridge's client-side code in order to mimic this default .NET behavior. The implementation has just been merged our integration branch and will be contemplated in the next Bridge release!

The next Bridge release containing this fix should reflect the milestone bound to this issue (the version may change as the code develops so we rely in the milestone attached to the issue to determine the version a given feature/fix went live).
